packages/ui/src/views/project/project-members/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { ChangeEvent, useCallback, useState } from 'react'
+
+import { Button, ListActions, NoData, PaginationComponent, SearchBox, SkeletonList, Spacer, Text } from '@/components'
+import { SandboxLayout } from '@/views'
+import { debounce } from 'lodash-es'
+
+import { InviteMemberDialog } from './components/invite-member-dialog'
+import { MembersList } from './components/member-list'
+import { MembersProps, ProjectMemberListViewProps } from './types'
+
+export const ProjectMemberListView: React.FC<ProjectMemberListViewProps> = ({
+  isLoading,
+  isInvitingMember,
+  useTranslationStore,
+  useMemberListStore,
+  usePrincipalListStore,
+  isInviteMemberDialogOpen,
+  setIsInviteMemberDialogOpen,
+  inviteMemberError,
+  searchQuery,
+  setSearchQuery,
+  onSubmit,
+  onEditMember
+}) => {
+  const { t } = useTranslationStore()
+  const { memberList, totalPages, page, setPage } = useMemberListStore()
+  const { principalList } = usePrincipalListStore()
+  const [searchInput, setSearchInput] = useState(searchQuery)
+
+  const debouncedSetSearchQuery = debounce(searchQuery => {
+    setSearchQuery(searchQuery || null)
+  }, 300)
+
+  const handleInputChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(e.target.value)
+    debouncedSetSearchQuery(e.target.value)
+  }, [])
+
+  const renderListContent = () => {
```

Comment:
possible to move this function out into it's own pure component?
this parent has a lot of props, and i assume it may be subject to frequent re-renders, causing this render function to get invalidated unnecessarily.
also, it looks cleaner to render `<ProjectMembersList props={} />` instead of `{ renderListContent(props) }`